app/src/main/java/org/openobservatory/ooniprobe/receiver/TestRunBroadRequestReceiver.java

```diff
@@ -0,0 +1,122 @@
+package org.openobservatory.ooniprobe.receiver;
+
+
+import android.content.BroadcastReceiver;
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.content.ServiceConnection;
+import android.os.IBinder;
+
+import com.google.common.collect.Lists;
+import com.google.common.math.Stats;
+
+import org.openobservatory.ooniprobe.common.PreferenceManager;
+import org.openobservatory.ooniprobe.common.service.RunTestService;
+import org.openobservatory.ooniprobe.test.TestAsyncTask;
+import org.openobservatory.ooniprobe.test.suite.AbstractSuite;
+
+import java.util.List;
+
+public class TestRunBroadRequestReceiver extends BroadcastReceiver implements ServiceConnection {
+    private final EventListener listener;
+    private final PreferenceManager preferenceManager;
+    public RunTestService service;
+    private boolean isBound;
+    private Integer runtime;
+
+    public TestRunBroadRequestReceiver(PreferenceManager preferenceManager, EventListener listener) {
+        this.preferenceManager = preferenceManager;
+        this.listener = listener;
+    }
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
+        String key = intent.getStringExtra("key");
+        String value = intent.getStringExtra("value");
+        switch (key) {
+            case TestAsyncTask.START:
+                listener.onStart(service);
+                break;
+            case TestAsyncTask.RUN:
+                listener.onRun(value);
+                break;
+            case TestAsyncTask.PRG:
+                List<AbstractSuite> previousTestSuits =
```

Comment:
I think the correct naming for the variable should be `previousTestSuites` (i.e., suites not suits)